src/prpy/util.py

```diff
@@ -749,3 +749,178 @@ def IsTimedTrajectory(trajectory):
     """
     cspec = trajectory.GetConfigurationSpecification()
     return cspec.ExtractDeltaTime(trajectory.GetWaypoint(0)) is not None
+
+
+def JointStatesFromTraj(robot, traj, times, derivatives=[0, 1, 2]):
+    """
+    Helper function to extract the joint position, velocity and acceleration
+    from an OpenRAVE trajectory.
+    @param robot The OpenRAVE robot
+    @param traj An OpenRAVE trajectory
+    @param times List of times in seconds
+    @param derivatives list of desired derivatives (only supports 0, 1, 2),
+                       defaults to [0, 1, 2]
+    @return pva_list List of list of position, velocity and acceleration
+                     at specified times.
+                     Inserts 'None' for unavailable or undesired fields
+                     The i-th element is the i-th derivative of position
+                     Of size |times| x 3
```

Comment:
This did fix my complaint, but it's incorrect with the new changes. The `i`-th element is the `derivative[i]`-th derivative of position and it's of size `|times| x |derivatives|`.